### PR TITLE
Patch sp plot

### LINKED ISSUE #71: A potential bug in stairway plot N(t) analysis?
When running the N(t) analysis on two demographic models "Constant" and Out-Of-Africa (YRI), @emanuelmfonseca and I noticed that the stairway plot results for these models are virtually identical (while msmc and GONE results do differ). @lauterbur noticed this on one of the calls where we showed the initial 1 rep result. We confirm that this happens consistently across different runs and on multiple replicates.

I also found that when running multiple replicates, the number of lines plotted for stairway plots using the current pipeline often don't match with the number of replicates specified in the config file, while msmc and GONE always do. Since `tiny_config.yaml` currently only have 1 rep and 1 demography these issues might be missed when debugging.

constant_all_estimated_Ne
OOF_all_estimated_Ne

### PR DESCRIPTION
**Pull request by andrewkern**
here is a patch for the bug that @lntran26 found and reported in #71, which appeared to be an issue with ploting stairwayplot results. 

In examining things, what was happening was that a helper function being used to expand the list of file names with stairwayplot results to plot was expanding over too many wildcards, and so inappropriate reps (e.g. between demographies) were being plotting together.

this PR provides a fix for this bug that will close #71

### CODE REVIEW

File: workflows/n_t.snake [hunk 1]
```diff
@@ -180,6 +180,7 @@ rule run_stairwayplot:
         rules.sp_download.output,
     output: output_dir + "/inference/sp/{demog}/{dfes}/{annots}/{seeds}/{chrms}/stairwayplot_estimated_Ne.txt"
     threads: 20
+    resources: mem_mb=120000
     run:
         inputs = expand(output_dir + "/simulated_data/{demog}/{dfes}/{annots}/{seeds}/sim_{chrms}.trees",
             seeds=seed_array,
```
Comment (lntran26) [inline workflows/n_t.snake:186]: Need to update line 186-190 to:
```
seeds=wildcards.seeds,
chrms=chrm_list,
dfes=wildcards.dfes,
demog=wildcards.demog,
annots=wildcards.annots,
```

File: workflows/n_t.snake [hunk 2]
```diff
@@ -208,8 +209,8 @@ def ne_files_sp(wildcards):
     return expand(output_dir + "/inference/sp/{demog}/{dfes}/{annots}/{seeds}/{chrms}/stairwayplot_estimated_Ne.txt",
         seeds=seed_array,
         chrms=chrm_list,
-        dfes=dfe_list,
-        demog=demo_model_ids,
+        dfes=wildcards.dfes,
+        demog=wildcards.demog,
         annots=annotation_list,
```
Comment (lntran26) [inline workflows/n_t.snake:214]: Also need to change line 214 to
`annots=wildcards.annots,`

**Review by lntran26 — CHANGES_REQUESTED** [review]
After making the additional suggested changes to the code I was able to generate stairway plots with the expected number of replicates for each of the two demographic models.

constant_all_estimated_Ne
OOF_all_estimated_Ne

**Comment by andrewkern** [comment]
thanks @lntran26 -- some of these edits didn't work for me. i've pushed further changes, fixes some other bugs that i saw, but am still trying to figure out whats going on with the SP output

**Comment by andrewkern** [comment]
okay, i can now confirm that this patch is working. here are representative plots from a run of the `config.yaml` settings. Both of these are Chr21, Gamma DFE on exons:

Constant demography, 
[all_estimated_Ne.pdf](https://github.com/popsim-consortium/analysis2/files/9355071/all_estimated_Ne.pdf)

OOA:
[all_estimated_Ne (1).pdf](https://github.com/popsim-consortium/analysis2/files/9355081/all_estimated_Ne.1.pdf)

**Comment by andrewkern** [comment]
i have rebased and squashed this PR so will merge.

**Pull request merged**